### ISSUE TITLE
[stable/rabbitmq] cannot log in after deployment

### ISSUE THREAD

**Issue by gboor** [helm/charts#7046]
**Is this a request for help?**: Yes

**Version of Helm and Kubernetes**:

```
Kubernetes: GKE with 1.9.7-gke.3, RBAC enabled
HELM: 2.9.1 - all charts up-to-date
```

**Which chart**: stable/rabbitmq

**What happened**:
Installation went fine, I provided username and password in the values.yaml. After deployment, it echoes back the correct username and gives me a command to retrieve the password, which also returns the correct password (both as set in the yaml file). I kubectl --proxy and go to the URL given by the chart (http://127.0.0.1:8002/api/v1/proxy/namespaces/default/services/rabbitmq:15672/), where I am presented with the RabbitMQ login screen, as expected.

When trying to log in with the username and password, it heads back to the login page, stating that login has failed, but also pops up an HTTP auth input box asking for credentials (as if Auth: basic is in place). Entering the credentials there does not work either. I cannot log in at all.

I have removed and re-deployed RabbitMQ numerous times, I have tried admin, user, the default passwords, etc., I dug into the logs and see that my user is "created successfully", but cannot get past the login screen.

Logs show no further errors. I tried installing the chart with debug enabled, but that does not work; I get a whole bunch of other errors flung at me which might belong in a different ticket.

**What you expected to happen**:
I expect to be able to log in to the admin console with the given username and password.

**How to reproduce it** (as minimally and precisely as possible):
1) Take the values-production.yaml
2) Set the username and password fields
3) Add some extra plugins. My full list is; rabbitmq_management,rabbitmq_peer_discovery_k8s,rabbitmq_stomp,rabbitmq_web_stomp,rabbitmq_mqtt
4) Deploy with helm install --name rabbitmq -f values-production.yaml
5) The output will give you a command for proxy with kubectl, run it
6) Go to the URL given in the output. May require some reloads before the underlying service is up and running, but it should present you with the RabbitMQ login screen
7) Try to log in, fail.

**Anything else we need to know**:
The fact that I get presented an auth basic login box after I try to log in is very strange. I am not sure if this behaviour is generated by RabbitMQ, or by something in GKE?

**Comment by wspi** [comment]
I've seen the same behaviour, but I'm not on GKE, running K8S via kops... and I don't even get the login failed message, whenever I try to login clicking on the button in the RabbitMQ login screen I get an Auth Basic that I can't get through. 

PS: I don't have any extra plugins installed

**Comment by gboor** [comment]
I can confirm the call to /api/whoami, which, according to the documentation, should return "details on the currently logged in user".

I inspected the request a bit more and saw that the www-authenticate header of the whoami call is set to

```
Basic realm="kubernetes-master"
```

Googling that gives me the Kubernetes API server source, not RabbitMQ. Could it be that calls to /api/whoami on RabbitMQ and being redirected to the Kubernetes API for some reason?

**Comment by gboor** [comment]
One further update; I just destroyed the service again (helm del --purge) and changed the serviceType in the values.yaml to LoadBalancer - essentially exposing the service through an external IP directly.

If I install it like that, it gives me some other command to get the external IP. It took a minute before that returned a valid IP, but once it did, I could access the console directly on that IP, on port 15672.

Using this approach I *CAN* log in and everything works as expected. This must really be some oddity that is specific to being proxied using kubectl, as directly accessing the service works.

This works, but it is far from ideal. While I have a sensible case to expose the AMQP port to the world, I really do not want the admin console to be accessible from outside the cluster... I know I can set up 2 services (a ClusterIP and a LoadBalancer) with the LoadBalancer only exposing AMQP, but while this bug exists, that means I cannot access the admin console at all.

**Comment by gboor** [comment]
More update;

This seems to be a direct result of this - https://github.com/kubernetes/kubernetes/issues/38775

Which means the experienced behaviour is "intended" for security reasons. It also means logging in to RabbitMQ via kubectl proxy is not possible at all and the instructions given at the end of the helm install are misleading.

**Comment by gboor** [comment]
This was inadvertently resolved by #7030, which gives port-forward commands that actually work. Closing this issue.
